mypy/checkexpr.py

```diff
@@ -58,6 +62,16 @@
 ArgChecker = Callable[[Type, Type, int, Type, int, int, CallableType, Context, MessageBuilder],
                       None]
 
+# Maximum nesting level for math union in overloads, setting this to large values
+# may cause performance issues.
```

Comment:
Maybe we should include add a blurb + a link telling people to read this PR if they want more details? Up to you.